### DISCUSSION TITLE
Adding the RichTextEditor to a custom module

### DISCUSSION THREAD

**Discussion by ryanjagdfeld** [oqtane/oqtane.framework#3962] [Q&A]
Hello!  First off, thank you for this awesome project and library, it has saved me a ton of extra work on most of my new projects.  I've been working on a project that has a RichTextEditor on one of my module's razor pages.  It was started in .Net 7 but I recently moved everything up to .Net 8 and the latest version of Oqtane.  After that, the RichTextEditor appears "weird" on the razor page.  All of the options are huge for the toolbar, and it stretches way down the page.   To troubleshoot the issue, I cloned a new Oqtane project, set it up using default render method and a local SQL server. I then created a new module from the latest template.  

I then added the CSS for quill to the Edit component resources:
```
    public override List<Resource> Resources => new List<Resource>()
    {
        new Resource { ResourceType = ResourceType.Stylesheet, Url = ModulePath() + "Module.css" },
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "css/quill/quill.bubble.css" },
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "css/quill/quill.snow.css" }
    };
```

Add a reference object:
```
    private RichTextEditor RichTextEditorHtml;
```

And then replaced the "Name" input with the RichTextEditor
```
    <div class="container">
        <div class="row mb-1 align-items-center">
            <Label Class="col-sm-3" For="name" HelpText="Enter a name" ResourceKey="Name">Name: </Label>
            <div class="col-sm-9">
                <RichTextEditor Content="@_name" AllowFileManagement="false" AllowRawHtml="false" @ref="@RichTextEditorHtml"></RichTextEditor>
            </div>
        </div>
    </div>
```

However, even after this, with a completely new project and new module, the editor still is not rendering correctly.  There are no errors in browser's console and all resources loaded correctly.  I can go to any of the Text/HTML modules on the pages, click on Edit and see the editor working as it should.  I feel like I'm just missing a reference to some CSS or something that is causing this.  Any help would be much appreciated!

Here is a screen shot of this new module with the RichTextEditor:
![Screenshot-RichTextEditor](https://github.com/oqtane/oqtane.framework/assets/111617998/3f09cb91-b02c-4bee-aadc-b89f49b26d89)

**Comment by sbwalker** [comment]
When you say "latest version" of Oqtane are you referring to the 5.0.2 release, or the 5.1 code in the Dev branch?

**Reply by ryanjagdfeld** [reply]
Ah ha!  I cloned the 5.1 right from the dev repo.  I tried it with 5.0.2 release, works fine.  Thank you!

**Reply by sbwalker** [reply]
Thank you for clarifying... I am able to reproduce in 5.1. will investigate further as it sounds like this is an issue which needs to be resolved before 5.1 is released.

**Reply by sbwalker** [reply]
@ryanjagdfeld this issue is now resolved with PR #3964 - thank you for reporting the problem!
